File: src/main/java/kr/mjc/rachel/spring/ioc/beanfactory/TVUser.java
package kr.mjc.rachel.spring.ioc.beanfactory;

/**
 * p.44 TVUser.java
 */
public class TVUser {
  public static void main(String[] args) {
    TV samsungTV = new SamsungTV();
    samsungTV.powerOn();
    samsungTV.powerOff();

    TV lgTV = new LgTV();
    lgTV.powerOn();
    lgTV.powerOff();
  }
}

File: src/main/kotlin/kr/mjc/jacob/spring/ioc/beanfactory/TVUser.kt
package kr.mjc.jacob.spring.ioc.beanfactory

/**
 * p.44 TVUser.java
 */
fun main() {
  SamsungTV().run {
    powerOn()
    powerOff()
  }

  LgTV().run {
    powerOn()
    powerOff()
  }
}